var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -24,161 +24,31 @@ class Warpx(CMakePackage, PythonExtension):
-    )
-    version(
-        "22.09",
-        sha256="dbef1318248c86c860cc47f7e18bbb0397818e3acdfb459e48075004bdaedea3",
-        deprecated=True,
-    )
-    version(
-        "22.08",
-        sha256="5ff7fd628e8bf615c1107e6c51bc55926f3ef2a076985444b889d292fecf56d4",
-        deprecated=True,
-    )
-    version(
-        "22.07",
-        sha256="0286adc788136cb78033cb1678d38d36e42265bcfd3d0c361a9bcc2cfcdf241b",
-        deprecated=True,
-    )
-    version(
-        "22.06",
-        sha256="e78398e215d3fc6bc5984f5d1c2ddeac290dcbc8a8e9d196e828ef6299187db9",
-        deprecated=True,
-    )
-    version(
-        "22.05",
-        sha256="2fa69e6a4db36459b67bf663e8fbf56191f6c8c25dc76301dbd02a36f9b50479",
-        deprecated=True,
-    )
-    version(
-        "22.04",
-        sha256="9234d12e28b323cb250d3d2cefee0b36246bd8a1d1eb48e386f41977251c028f",
-        deprecated=True,
-    )
-    version(
-        "22.03",
-        sha256="ddbef760c8000f2f827dfb097ca3359e7aecbea8766bec5c3a91ee28d3641564",
-        deprecated=True,
-    )
-    version(
-        "22.02",
-        sha256="d74b593d6f396e037970c5fbe10c2e5d71d557a99c97d40e4255226bc6c26e42",
-        deprecated=True,
-    )
-    version(
-        "22.01",
-        sha256="e465ffadabb7dc360c63c4d3862dc08082b5b0e77923d3fb05570408748b0d28",
-        deprecated=True,
-    )
     # 22.01+ requires C++17 or newer
-    version(
-        "21.12",
-        sha256="847c98aac20c73d94c823378803c82be9a14139f1c14ea483757229b452ce4c1",
-        deprecated=True,
-    )
-    version(
-        "21.11",
-        sha256="ce60377771c732033a77351cd3500b24b5d14b54a5adc7a622767b9251c10d0b",
-        deprecated=True,
-    )
-    version(
-        "21.10",
-        sha256="d372c573f0360094d5982d64eceeb0149d6620eb75e8fdbfdc6777f3328fb454",
-        deprecated=True,
-    )
-    version(
-        "21.09",
-        sha256="861a65f11846541c803564db133c8678b9e8779e69902ef1637b21399d257eab",
-        deprecated=True,
-    )
-    version(
-        "21.08",
-        sha256="6128a32cfd075bc63d08eebea6d4f62d33ce0570f4fd72330a71023ceacccc86",
-        deprecated=True,
-    )
-    version(
-        "21.07",
-        sha256="a8740316d813c365715f7471201499905798b50bd94950d33f1bd91478d49561",
-        deprecated=True,
-    )
-    version(
-        "21.06",
-        sha256="a26039dc4061da45e779dd5002467c67a533fc08d30841e01e7abb3a890fbe30",
-        deprecated=True,
+    # 20.01+ requires C++14 or newer
+
+    # CMake: Fix List of Pip Options
+    patch(
+        "https://github.com/ECP-WarpX/WarpX/pull/5378.patch?full_index=1",
+        sha256="7357a1001426d5eb633df993b2a719f738e8073f08876c4a90e32b1d4f11ebea",
+        when="@24.10",
     )
-    version(
-        "21.05",
-        sha256="f835f0ae6c5702550d23191aa0bb0722f981abb1460410e3d8952bc3d945a9fc",
-        deprecated=True,
+
+    # CMake: No FFTW Needed for SYCL anymore
+    patch(
+        "https://github.com/ECP-WarpX/WarpX/pull/5380.patch?full_index=1",
```

Comment:
For this to work we will also need to patch in https://github.com/ECP-WarpX/WarpX/pull/5127
```suggestion
    patch(
        "https://github.com/ECP-WarpX/WarpX/pull/5127.patch?full_index=1",
        sha256="d2e112371582435399682bf7c1498e6ec04a36d62b2dbce329d566b862074c16",
        when="@24.10",
    )
    patch(
        "https://github.com/ECP-WarpX/WarpX/pull/5380.patch?full_index=1",
```